exo-deploy/src/deployers/aws-deployer.ls

```diff
@@ -39,20 +41,62 @@ class AwsDeployer
 
   deploy: ->
     @terraform
-      ..get (err) ->
+      ..get (err) ~>
         | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting deploy to AWS"
-        ..apply (err) ->
-          | err => return process.stdout.write err.message
+        @_get-hosted-zone-id destroy: no, (err, hosted-zone-id) ~>
+          | err => process.stdout.write "Cannot get hosted zone id #{err.message}" ; return
+          ..apply {hosted-zone-id}, (err) ->
+            | err => return process.stdout.write err.message
 
 
   nuke: ->
-    @terraform
-      ..get (err) ->
-        | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting nuke from AWS"
-        ..destroy (err) ~>
-          | err => return process.stdout.write err.message
+    @terraform-file-builder.generate-provider-credentials!
+    process.stdout.write "removing the entire AWS deployment"
+    @_get-hosted-zone-id destroy: yes, (err, hosted-zone-id) ~>
+      | err => process.stdout.write "Cannot get hosted zone id #{err.message}" ; return
+      @terraform.destroy {hosted-zone-id}, (err) ~>
+          | err => process.stdout.write "Terraform cannot destroy infrastructure #{err.message}" ; return
+          @_remove-hosted-zone @domain-name, (err) ->
+            | err => process.stdout.write "Cannot remove hosted zone #{err.message}"
+
+
+  _get-hosted-zone-id: ({destroy}, done) ->
+    @_hosted-zone-exists @domain-name, (err, hosted-zone) ~>
+      | err         => process.stdout.write err.message ; return done err
+      | hosted-zone => done null, hosted-zone.Id
+      | _           =>
+        if !destroy
+          @_create-hosted-zone @domain-name, (err, hosted-zone) ->
+            | err => process.stdout.write "Cannot create hosted zone #{err.message}" ; return done err
+            done null, hosted-zone.Id
+        else done null, null
+
+
+  _hosted-zone-exists: (domain-name, done) ->
+    @route53 = new Aws.Route53 api-version: '2013-04-01'
+      ..list-hosted-zones null, (err, data) ~>
+        | err => process.stdout.write "Cannot list hosted zones #{err.message}" ; return done err
+        done null, (data.HostedZones |> find (.Name is "#{domain-name}."))
+
+
+  _create-hosted-zone: (domain-name, done) ->
+    params =
+      CallerReference: uuid.v4!
+      Name: domain-name
+
+    @route53.create-hosted-zone params, (err, data) ~>
+      | err => process.stdout.write err.message ; return done err
+      process.stdout.write "Please add the following name servers to #{@domain-name}:\n"
+      for name-server in data.DelegationSet.NameServers
+        process.stdout.write "#{name-server}\n"
+      done null, data.HostedZone
+
+
+  _remove-hosted-zone: (domain-name) ->
```

Comment:
You call this method with a continuation callback, but it doesn't accept one. Please add one here.